bmp388/bmp388.go

```diff
@@ -0,0 +1,246 @@
+	if err != nil {
+		return 0, err
+	}
+	rawPress, err := d.readSensorData(REG_PRESS)
+	if err != nil {
+		return 0, err
+	}
+
+	// code pulled from bmp388 C driver: https://github.com/BoschSensortec/BMP3-Sensor-API/blob/master/bmp3.c
+	partialData1 := d.cali.tlin * d.cali.tlin
+	partialData2 := partialData1 / 64
+	partialData3 := (partialData2 * d.cali.tlin) / 256
+	partialData4 := (int64(d.cali.p8) * partialData3) / 32
+	partialData5 := (int64(d.cali.p7) * partialData1) * 16
+	partialData6 := (int64(d.cali.p6) * d.cali.tlin) * 4194304
+	offset := (int64(d.cali.p5) * 140737488355328) + partialData4 + partialData5 + partialData6
+	partialData2 = (int64(d.cali.p4) * partialData3) / 32
+	partialData4 = (int64(d.cali.p3) * partialData1) * 4
+	partialData5 = (int64(d.cali.p2) - 16384) * d.cali.tlin * 2097152
+	sensitivity := ((int64(d.cali.p1) - 16384) * 70368744177664) + partialData2 + partialData4 + partialData5
+	partialData1 = (sensitivity / 16777216) * rawPress
+	partialData2 = int64(d.cali.p10) * d.cali.tlin
+	partialData3 = partialData2 + (65536 * int64(d.cali.p9))
+	partialData4 = (partialData3 * rawPress) / 8192
+
+	// dividing by 10 followed by multiplying by 10
+	// To avoid overflow caused by (pressure * partial_data4)
+	partialData5 = (rawPress * (partialData4 / 10)) / 512
+	partialData5 = partialData5 * 10
+	partialData6 = (int64)(uint64(rawPress) * uint64(rawPress))
+	partialData2 = (int64(d.cali.p11) * partialData6) / 65536
+	partialData3 = (partialData2 * rawPress) / 128
+	partialData4 = (offset / 4) + partialData1 + partialData5 + partialData3
+	compPress := ((uint64(partialData4) * 25) / uint64(1099511627776))
+	return int32(compPress), nil
+}
+
+// ReadAltitude estimates the altitude above sea level in centimeters.
+func (d *Device) ReadAltitude() (alt int32, err error) {
+	press, err := d.ReadPressure()
+	if err != nil {
+		return
+	}
+	// pulled from bme280 driver, modified a bit to return altitude in centimeters
+	alt = int32(4433000.0 * (1.0 - math.Pow(float64(press)/float64(d.Config.SeaLevelPressure), 0.1903)))
+	return
+}
+
+// SoftReset commands the BMP388 to reset of all user configuration settings
+func (d *Device) SoftReset() error {
+	err := d.writeRegister(REG_CMD, SOFT_RESET)
+	if err != nil {
+		return errors.New("bmp388: failed to perform a soft reset")
+	}
+	return nil
+}
+
+// Connected tries to reach the bmp388 and check its chip id register. Returns true if it was able to successfully
+// communicate over i2c and returns the correct value
+func (d *Device) Connected() bool {
+	data, err := d.readRegister(REG_CHIP_ID, 1)
+	return err == nil && data[0] == CHIP_ID // returns true if i2c comm was good and response equals 0x50
+}
+
+// SetMode changes the run mode of the sensor, NORMAL is the one to use for most cases. Use FORCED if you plan to take
+// measurements infrequently and want to conserve power. SLEEP will of course put the sensor to sleep
+func (d *Device) SetMode(mode Mode) error {
```

Comment:
We might eventually want to standardize this sort of thing, but for now it's okay.